benchmarks/wrappers.py

```diff
@@ -0,0 +1,158 @@
+            del self.ctx
+
+    def time_resize_smaller(self, kind):
+        self.assert_clean_setup()
+        self.ds.resize((10, 10, 10))
+        if kind != "h5py.Dataset":
+            self.ctx.__exit__(None, None, None)  # commit
+            del self.ctx
+
+    def teardown(self, kind):
+        if hasattr(self, "ctx"):
+            self.ctx.__exit__(None, None, None)  # commit
+            del self.ctx
+        super().teardown()
+
+
+class TimeCreateDataset(Benchmark):
+    params = [["dense", "sparse"], ["h5py", "versioned_hdf5"]]
+    param_names = ["density", "library"]
+
+    def setup(self, density, library):
+        super().setup()
+        self.kwargs = {"chunks": (3, 3, 3)}
+        if density == "dense":
+            self.kwargs["data"] = np.arange(10000).reshape((100, 10, 10))
+        else:
+            self.kwargs["shape"] = (100, 10, 10)
+
+    def time_create_dataset(self, density, library):
+        self.assert_clean_setup()
+        if library == "h5py":
+            self.file.create_dataset("data", **self.kwargs)
+        else:
+            with self.vfile.stage_version("v0") as sv:
+                sv.create_dataset("data", **self.kwargs)
+        self.file.close()
+
+
+class TimeCommit(Benchmark):
+    params = [
+        [
+            "v1_dense",
+            "v1_sparse",
+            "v2_no_changes",
+            "v2_modified_no_changes",
+            "v2_modified_all_changes",
+        ],
+        [(10, 10), (50, 50), (250, 250)],
+    ]
+    param_names = ["kind", "chunks"]
+
+    def setup(self, kind, chunks):
+        shape = (500, 500)
+        data = np.arange(500 * 500).reshape(shape)
+
+        super().setup()
+        if kind.startswith("v1_"):
+            self.ctx = self.vfile.stage_version("v1")
+            self.version = self.ctx.__enter__()
+            if kind == "v1_dense":
+                self.version.create_dataset("data", data=data, chunks=chunks)
+            elif kind == "v1_sparse":
+                self.version.create_dataset("data", shape=shape, chunks=chunks)
+            else:
+                raise AssertionError("unreachable")
+
+        elif kind.startswith("v2_"):
+            with self.vfile.stage_version("v1") as version:
+                version.create_dataset("data", data=data, chunks=chunks)
+            self.ctx = self.vfile.stage_version("v2")
+            self.version = self.ctx.__enter__()
+            ds = self.version["data"]
+            if kind == "v2_no_changes":
+                pass
+            elif kind == "v2_modified_no_changes":
+                ds[:] = ds[:]
+            elif kind == "v2_modified_all_changes":
+                ds[:] = ds[:] + 123
+            else:
+                raise AssertionError("unreachable")
+
+        else:
+            raise AssertionError("unreachable")
+
+    def time_commit(self, kind, chunks):
+        self.assert_clean_setup()
+        self.ctx.__exit__(None, None, None)
+        self.file.close()
```

Comment:
Should this go into a `teardown` method? Does it even contribute much overhead at all, or is it negligible?

Reply:
You're right, it's inconsequential